R/print.data.table.R

```diff
@@ -64,15 +65,27 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
     }
     return(invisible(x))
   }
+  index_dt = NULL
+  if (show.indices) {
+    if (is.null(indices(x))) {
+      show.indices = FALSE
+    } else {
+      index_dt <- as.data.table(attributes(attr(x, 'index')))
+      print_names <- paste0("index", if (ncol(index_dt) > 1L) seq_len(ncol(index_dt)) else "", ":", sub("^__", "", names(index_dt), fixed=TRUE))
```

Comment:
I'm wondering if it's worth maintaining the complexity of differentiating the n_index >1 vs =1 cases. Without it we can simplify:

```r
sprintf("index%d:%s", seq_len(ncol(index_dt)), sub("^__", "", names(index_dt)))
```

Reply:
IMO the distinction feels a bit more polished and to keep it doesn't introduce a whole lot of complexity, so I'd rather keep it